internal/storage/fs/snapshot.go

```diff
@@ -614,7 +617,16 @@ func (ss *Snapshot) addDoc(doc *ext.Document) error {
 
 		ns.evalRollouts[f.Key] = evalRollouts
 	}
-	ns.etag = doc.Etag
+	etag := doc.Etag
+	if ns.etag != "" {
+		hash := sha1.New() //nolint:gosec
+		hash.Write([]byte(ns.etag))
+		hash.Write([]byte(etag))
+		sum := hash.Sum(nil)
+		etag = hex.EncodeToString(sum)
+		etag = etag[:12] // truncate to 12 characters for consistency
```

Comment:
just curious why 12 and not something larger like 32?

Reply:
#4501

Reply:
@markphelps Docker and Git typically display a shortened version of the hash. Docker, for example, uses 12 characters. We don't use it for security reason so I decided to have a short version of it